compiler/rustc_const_eval/src/interpret/simd_intrinsics.rs

```diff
@@ -0,0 +1,811 @@
+use either::Either;
+use rustc_abi::{Endian, HasDataLayout, Size};
+use rustc_apfloat::Float;
+use rustc_apfloat::ieee::{Double, Half, Quad, Single};
+use rustc_middle::mir::interpret::{
+    InterpErrorKind, InterpResult, Provenance, UndefinedBehaviorInfo, interp_ok,
+};
+use rustc_middle::ty::FloatTy;
+use rustc_middle::{bug, err_ub_format, mir, span_bug, throw_ub_format, throw_unsup_format, ty};
+use rustc_span::{Symbol, sym};
+use tracing::trace;
+
+use super::{ImmTy, InterpCx, Machine, OpTy, PlaceTy, Scalar};
+use crate::interpret::Writeable;
+use crate::interpret::intrinsics::MinMax;
+
+fn check_intrinsic_arg_count<'a, 'tcx, P, const N: usize>(
+    args: &'a [OpTy<'tcx, P>],
+) -> InterpResult<'tcx, &'a [OpTy<'tcx, P>; N]>
+where
+    &'a [OpTy<'tcx, P>; N]: TryFrom<&'a [OpTy<'tcx, P>]>,
+    P: Provenance,
```

Comment:
This is a Miri function, we don't do this for intrinsics in the core interpreter. Please port these to the usual convention we use here.

(Alternatively we could start using this in the core interpreter, but then a separate PR should first port the existing intrinsics.)